.github/workflows/tests.yml

```diff
@@ -12,23 +12,13 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v4
-      - uses: bahmutov/npm-install@v1
+      - uses: bahmutov/npm-install@HEAD
         with:
           working-directory: frontend
-      - name: Build docker image
-        run: ./frontend/docker/build
-
-      - name: Build and run backend docker
-        run: |
-          docker-compose run api bundle exec rake db:migrate
-          docker-compose up -d
-      - name: log docker status
+      - name: Build and run docker
         run: |
-          sleep 30
-          docker ps -a
-          docker logs kinetic_api_1
-          curl -i http://localhost:4006/api/v1/openapi.json
-          docker logs kinetic_api_1
+          docker compose run api bundle exec rake db:migrate
+          docker compose up -d
```

Comment:
Yeah this should be faster overall. The rspec job is basically the same amount of time ~3 minutes. 

For the rubocop job, that was previously building the backend docker container and running rubocop inside of that which [took about ~1 minute if files were changed](https://github.com/openstax/kinetic/actions/runs/7415500430/job/20178736597?pr=336)

[if no files were changed, it still took around ~20-30 seconds](https://github.com/openstax/kinetic/actions/runs/7422695189/job/20198543671?pr=335)

with these changes it [gets us down to around ~11 seconds](https://github.com/openstax/kinetic/actions/runs/7422429232/job/20197694979?pr=325) (at least on this pr) and that includes scanning all 200+ files, we could try to further optimize and emulate what we were doing by only checking files that have changed in the PR, however i feel like it's nice to have the full coverage (especially since its so quick) and in case we change the target ruby version without touching any other files, this would cover that case as well